### PR TITLE
[MM-27836] Fix possible panic when patching config

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes two instances where sending an `nil` `SiteURL` could cause a panic when patching the config.

#### Ticket

https://mattermost.atlassian.net/browse/MM-27836

### CODE REVIEW

File: api4/config.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	appCfg := c.App.Config()
-	if *appCfg.ServiceSettings.SiteURL != "" && *cfg.ServiceSettings.SiteURL == "" {
+	if *appCfg.ServiceSettings.SiteURL != "" && (cfg.ServiceSettings.SiteURL == nil || *cfg.ServiceSettings.SiteURL == "") {
```
Comment (ashishbhate) [inline api4/config.go:82]: How is it that `cfg.ServiceSettings.SiteURL` can be `nil` here? On line 74 we call `cfg.SetDefaults` which calls `ServiceSettings.SetDefaults` which ensures `SiteURL` wont be nil.

Comment (streamer45) [reply api4/config.go:82]: I forgot to update the method and saw the test fail but you are right, only the patch has the issue. Let me update.

**Comment by amyblais** [comment]
@streamer45 Kind note to cherry-pick this,

**Pull request merged**